hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -1183,12 +1191,26 @@ AmazonS3 getAmazonS3Client() {
    * @param reason a justification for requesting access.
    * @return AmazonS3Client
    */
+  // TODO: Remove when we remove S3V1 client
   @VisibleForTesting
   public AmazonS3 getAmazonS3ClientForTesting(String reason) {
     LOG.warn("Access to S3A client requested, reason {}", reason);
     return s3;
   }
 
+  /**
+   * Returns the S3 client used by this filesystem.
+   * <i>Warning: this must only be used for testing, as it bypasses core
+   * S3A operations. </i>
+   * @param reason a justification for requesting access.
+   * @return S3Client
+   */
+  @VisibleForTesting
+  public S3Client getAmazonS3V2ClientForTesting(String reason) {
+    LOG.warn("Access to S3A client requested, reason {}", reason);
```

Comment:
while we are here...

```suggestion
    LOG.warn("Access to S3 client requested, reason {}", reason);
```